cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java

```diff
@@ -289,7 +290,23 @@ public CassandraDaemonTask updateId() {
 
     @Nullable
     private static DiscoveryInfo getDiscoveryInfo(
-            Capabilities capabilities, String nodeName, int nativePort) {
+            boolean publishDiscoveryInfo, String clusterName, Capabilities capabilities, String nodeName,
+            int nativePort) {
+
+        // If the explicit configuration flag for publishing discovery info is set, include the cluster name in the
+        // discovery info name and don't use labels.
+        if (publishDiscoveryInfo) {
+            LOGGER.info("Publishing discovery info: ")
+            DiscoveryInfo.Builder discoveryBuilder = DiscoveryInfo.newBuilder()
+                    .setVisibility(DiscoveryInfo.Visibility.EXTERNAL)
+                    .setName(clusterName + "." + nodeName);
+            discoveryBuilder.getPortsBuilder().addPortsBuilder()
+                    .setName("NativeTransport")
+                    .setNumber(nativePort);
+            return discoveryBuilder.build();
+        }
+
```

Comment:
No.  The `VIP_NODE_PORT` is the port that clients should use to communicate with the VIP.  It should not be allocated by Mesos.  The nativePort is the port allocated by Mesos.  You can make up whatever port you like for `VIP_NODE_PORT`, minuteman maps it to `nativePort`.